src/brpc/policy/baidu_rpc_protocol.cpp

```diff
@@ -642,12 +642,16 @@ void ProcessRpcRequest(InputMessageBase* msg_base) {
         bthread_assign_data((void*)&server->thread_local_options());
     }
 
-    Span* span = NULL;
+    std::shared_ptr<Span> span;
     if (IsTraceable(request_meta.has_trace_id())) {
         span = Span::CreateServerSpan(
             request_meta.trace_id(), request_meta.span_id(),
             request_meta.parent_span_id(), msg->base_real_us());
+#if BRPC_SPAN_ENABLE_SHARED_PTR_API
         accessor.set_span(span);
+#else
+        accessor.set_span(span.get());
```

Comment:
you can add an overloaded function `set_span(std::shared_ptr<Span> span)`,
so that you don't need to use `BRPC_SPAN_ENABLE_SHARED_PTR_API` everywhere.